src/main/java/com/proofpoint/discovery/DynamicStore.java

```diff
@@ -25,4 +27,6 @@
     Stream<Service> getAll();
     Stream<Service> get(String type);
     Stream<Service> get(String type, String pool);
+
+    Entry get(Id<Node> nodeId);
```

Comment:
Is `Entry` really the correct return type here? `Entry` is a bit specific to the replication protocol. This change has to  add code to `InMemoryDynamicStore` to construct an `Entry` to return from this method, but the callers only seem to ever call `getAnnouncer()` on it.